web/views/bundles_view.ex

```diff
@@ -1,92 +1,45 @@
+    enabled_version = if Ecto.assoc_loaded?(enabled_version) do
+      %{enabled_version: render_one(enabled_version, BundleVersionView, "bundle_version.json", as: :bundle_version)}
+    else
+      %{}
+    end
+
+    relay_groups = if Ecto.assoc_loaded?(bundle.relay_groups) do
+      %{relay_groups: render_many(bundle.relay_groups, RelayGroupView, "relay_group.json", as: :relay_group)}
+    else
+      %{}
+    end
 
-  defp ordered_version_strings(versions) do
-    versions
-    |> Enum.map(&(&1.version))
-    |> Enum.sort
-    |> Enum.map(&to_string/1)
-  end
-
-
-  def render("bundle.json", %{bundle: bundle, enabled_bundles: enabled_bundles}=_params) do
-
-    enabled_version = case Map.get(enabled_bundles, bundle.name) do
-                        nil -> %{}
-                        version -> %{enabled_version: to_string(version)}
-                      end
     %{id: bundle.id,
       name: bundle.name,
-      versions: ordered_version_strings(bundle.versions),
+      versions: ordered_versions(bundle.versions),
       inserted_at: bundle.inserted_at,
-      updated_at: bundle.updated_at,
-      relay_groups: []}
+      updated_at: bundle.updated_at}
     |> Map.merge(enabled_version)
-
+    |> Map.merge(relay_groups)
   end
-
-  def render("index.json", %{bundles: bundles}=assigns) do
-    %{bundles: render_many(bundles, __MODULE__,
-                           "bundle.json",
-                           Map.put(assigns, :as, :bundle))}
-  end
-
-  def render("show.json", %{bundle: bundle, warnings: warnings}=assigns)
-  when length(warnings) > 0 do
-    warnings = Enum.map(warnings, fn({msg, meta}) -> ~s(Warning near #{meta}: #{msg}) end)
-    %{warnings: warnings,
-      bundle: render_one(bundle, __MODULE__, "bundle.json",
-                         Map.merge(assigns,
-                                   %{as: :bundle,
-                                     include: [:commands, :relay_groups, :namespace]}))}
+  def render("index.json", %{bundles: bundles}) do
+    %{bundles: render_many(bundles, __MODULE__, "bundle.json", as: :bundle)}
   end
-  def render("show.json", %{bundle: bundle}=assigns) do
-    %{bundle: render_one(bundle, __MODULE__,
-                         "bundle.json",
-                         Map.merge(assigns,
-                                   %{as: :bundle,
-                                     include: [:commands, :relay_groups, :namespace]}))}
+  def render("show.json", %{bundle: bundle}) do
+    %{bundle: render_one(bundle, __MODULE__, "bundle.json", as: :bundle)}
   end
 
-  # defp render_includes(inc_fields, resource) do
-  #   Map.get(inc_fields, :include, [])
-  #   |> Enum.reduce(%{}, fn(field, reply) ->
-  #     case render_include(field, resource) do
-  #       nil -> reply
-  #       {key, value} -> Map.put(reply, key, value)
-  #     end
-  #   end)
-  # end
+  ########################################################################
+
+  defp ordered_versions(versions) do
```

Comment:
We actually are using Elixir's `Version` for our custom Ecto `VersionTriple` type. Perhaps I should have named this variable `bundle_versions` instead of `versions`, since they're the models.